core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -414,9 +414,7 @@ class CudaInternal {
     return cudaStreamDestroy(stream);
   }
 
-  template <bool setCudaDevice = true>
   cudaError_t cuda_stream_synchronize_wrapper(cudaStream_t stream) const {
-    if constexpr (setCudaDevice) set_cuda_device();
```

Comment:
There is no need for the wrapper at this point. Other places the cuda backend directly call cuda API so might as well with `cudaStreamSynchronize`. It's only call in a single place I think. Maybe others disagree?

Reply:
Indeed, let's just clean this :+1: